README.md

```diff
@@ -2,208 +2,394 @@
+them both available in the global scope. If we were to continue adding code to
+this file, we could access both of them anywhere we want.
+
+While global scope makes it easy to access variables and functions wherever and
+whenever we want to, it is not always a good idea. In general, it's best
+practice to make variables and functions available only where they're needed
+— and nowhere else.
+
+Making a variable available in places that shouldn't have access to it can lead
+to bad things and make your debugging process more complex. The more pieces of
+code that can access a given variable, the more places you have to check for
+bugs if/when that variable contains an unexpected value.
+
+This is where the other levels of scope comes in handy.
+
+## Function Scope
+
+Function scope is the **scope within a single function**. Anything **defined
+inside the body of a function** is considered to be part of that function's
+scope.
+
+Anything defined in the function's scope can be accessed anywhere _within the
+function's body_. For example:
 
 ```js
-function myFunc() {
-  const myVar = 42;
+let count = 0;
+
+function counter() {
+  const resetValue = 100;
+  console.log("Counter will reset after: ", resetValue);
+
+  if (count < resetValue) {
+    count++;
+  } else {
+    count = 0;
+  }
 }
-// => undefined
 
-myVar * 2;
-// Uncaught ReferenceError: myVar is not defined
+counter(); // => Counter will reset after: 100
 ```
 
-The function body creates its own scope. It's like a separate channel on Slack
-— only the members of that channel can read the messages sent in it.
+In the above example, the `resetValue` variable is defined within the `counter`
+_function's scope_. We can see it's accessible anywhere within the function,
+both in the `console.log` and the `if` statement.
 
-## Block scope
+The example also showcases again how `count`, a variable defined in the _global
+scope_, is accessible anywhere in our code.
 
-A block statement also creates its own scope... kind of.
+The reverse is not true, however. Variables defined in the _function scope_ are
+**not** accessible anywhere outside of that scope. In other words, the global
+scope cannot access the function scope.
 
-Variables declared with `var` are **not** block-scoped:
+Let's take a look at a modified version of our `counter` example to see that in
+action:
 
 ```js
-if (true) {
-  var myVar = 42;
+let count = 0;
+function counter() {
+  const resetValue = 100;
+  console.log("Counter will reset after: ", resetValue);
 }
 
-myVar;
-// => 42
+counter(); // => Counter will reset after: 100
+console.log(resetValue); // => ReferenceError: resetValue is not defined
 ```
 
-However, variables declared with `const` and `let` **are** block-scoped:
+The `resetValue` variable does not exist outside of its function's scope.
+Anything outside of a function cannot reference anything defined within it.
 
-```js
-if (true) {
-  const myVar = 42;
+## Block Scope
 
```

Comment:
Do we define what a code "block" is elsewhere? Could we write a brief synopsis of that here? Even if we address it elsewhere, redundancy helps concepts stick.